### PR TITLE
Refactor theme fonts class for readability

### PR DESCRIPTION
**Pull request by vcanales**
## What?
This Pull Request refactors the them fonts class to improve code readability and maintainability. The main focus is on cleaning up the code.

I felt inclined to do this because when debugging for https://github.com/WordPress/create-block-theme/pull/660, I found some parts of the code hard to follow.

## Why?
This PR addresses the need to make the code more readable and maintainable by:
1. Adding documentation and comments.
2. Simplifying functions and improving their readability.

### Key Changes:
1. **Class documentation added**: Added descriptions and the purpose of the `CBT_Theme_Fonts` class.

2. **Function improvements**:
   - *Inline Descriptions/Justifications*:
     - Refactored `make_theme_font_src_absolute()` to use `array_map` for cleaner array handling.
     - Replaced double newlines and improved spacing for better readability.
   - Changed `foreach` to `array_map`: This reduces the complexity and makes the handling of arrays cleaner.

3. **Code clean-up**:
   - Removed some redundant comments in favor of "self documenting" practices.
   - Used shorthand assignment (`?? null`) for checking array keys.
   - Introduced variables for commonly used expressions.

4. **Function calls**:
   - Replaced `CBT_Theme_Fonts::` with `self::` for static method calls within the class, enhancing readability and context awareness.

5. **Added test cases** for parts of the code that could break on further refactors.

### CODE REVIEW

**Review by matiasbenedetto — COMMENTED** [review]
Probably we should use this refactor to improve the tests of this class. They currently relay on setting users, creating new themes dynamically and after removing it without restoring the previous state.

Example:

https://github.com/WordPress/create-block-theme/blob/eabd38201d6a7cbb1631be4cf9eae820fe838028/tests/test-theme-fonts.php#L26-L60

 I think this kind of test is the cause of the destructive tests we found some time ago and we tried to workaround it here: https://github.com/WordPress/create-block-theme/pull/618

Could we follow the approach we are using in the [base case for the Readme class](https://github.com/WordPress/create-block-theme/blob/trunk/tests/CbtThemeReadme/base.php) ?. It can improve 3 aspects of these tests:

- Fine tune control and always predictable active theme for test.
- No destructive test env.
- Avoid realying on CBT complex functions.

File: tests/test-theme-fonts.php [hunk 1]
```diff
@@ -214,6 +264,117 @@ public function test_get_all_fonts_from_theme_and_variation() {
...
+		// Verify that the src is correctly handled
+		$this->assertCount( 1, $fonts );
+		$this->assertEquals( 'single-src-font', $fonts[0]['slug'] );
+		$this->assertEquals( get_stylesheet_directory_uri() . '/assets/fonts/single-src-font.ttf', $fonts[0]['fontFace'][0]['src'] );
+
+		$this->uninstall_theme( $test_theme_slug );
+	}
+
+	public function test_array_font_src() {
+		wp_set_current_user( self::$admin_id );
```
Comment (matiasbenedetto) [inline tests/test-theme-fonts.php:304]: This is probably not needed.

Comment (vcanales) [reply tests/test-theme-fonts.php:304]: Without it, the tests break:

```php
Time: 00:00.357, Memory: 42.50 MB

There was 1 error:

1) Test_Create_Block_Theme_Fonts::test_copy_activated_fonts_to_theme
Undefined array key "typography"

/var/www/html/wp-content/plugins/create-block-theme/tests/test-theme-fonts.php:43

ERRORS!
Tests: 67, Assertions: 163, Errors: 1.
Script phpunit handling the test event returned with error code 2
✖ Command failed with exit code 2
Command failed with exit code 2
```

Comment (vcanales) [reply tests/test-theme-fonts.php:304]: > Probably we should use this refactor to improve the tests of this class. They currently relay on setting users, creating new themes dynamically and after removing it without restoring the previous state.

I agree that we should work on that, but since the tests currently cover the already-refactored code, I'd prefer refactoring them in another PR.

File: includes/create-theme/theme-fonts.php [hunk 2]
```diff
@@ -74,53 +72,42 @@ public static function get_all_fonts() {
...
 
 	public static function copy_activated_fonts_to_theme() {
-		$user_settings = CBT_Theme_JSON_Resolver::get_user_data()->get_settings();
-		if ( ! isset( $user_settings['typography']['fontFamilies']['custom'] ) ) {
-			return null;
-		}
-
-		$font_families_to_copy = $user_settings['typography']['fontFamilies']['custom'];
+		$user_settings         = CBT_Theme_JSON_Resolver::get_user_data()->get_settings();
+		$font_families_to_copy = $user_settings['typography']['fontFamilies']['custom'] ?? null;
```
Comment (pbking) [inline includes/create-theme/theme-fonts.php]:
```suggestion
		$font_families_to_copy = self::get_user_activated_fonts();
```

**Review by pbking — APPROVED** [review]
Seems fine to me.  Quality improvements.
